card/mod/core/set/all/actify.rb

```diff
@@ -31,25 +32,16 @@ def start_new_act opts
   ActManager.clear
   self.director = nil
   ActManager.act_card = self
-  if opts && opts[:success]
-    Env[:success] = Env::Success.new(name, Env.params[:success])
-  end
-  run_callbacks :act do
-    yield
-  end
+  Env.success(name) if opts[:success]
```

Comment:
The not nil check was probably there for reason (just because the default is a hash doesn't mean that it can't be nil). If you remove it make at least sure that it is a hash. An `opts ||= {}` at the beginning of the method will probably save us a ton of airbrake messages and a debugging chore.